app/Http/Controllers/AdminUserController.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Helpers\Responder;
+use App\Model\AdminUser;
+use App\Model\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+
+class AdminUserController extends Controller
+{
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * @param Request $request
+     * 后台登录
+     * @return array
+     */
+    public function login(Request $request){
+        // step 1. 验证数据
+        $account = $request->input('account');
+        $password = $request->input('password');
+
+        //  验证数据
+        $validator = Validator::make($request->all(), [
+            'account' => ['required'],
+            'password' => ['required'],
+        ],[
+            'account.required' => '学号不能为空',
+            'password.required' => '密码不能为空',
+        ]);
+
+        if($validator->fails()){
+           return Responder::error('0001',$validator->errors()->first());
+        }
+
+        //step 2. 判断用户是否存在
+        $result = AdminUser::where('account',$account)->exists();
+        if(!$result){
+           return Responder::error('0002','账号不正确');
+        }
+
+        //step 3. 判断密码是否正确
+        /**
+         * $DBpassword：数据库中拿出来的密码
+         * $result:验证数据库跟输入密码是否一致
+         */
+
+        $DBpassword = AdminUser::select('password')->first()->password;
+        $result = password_verify($password,$DBpassword);
+
+        if(!$result){
+            return Responder::error('0003','密码不正确');
+        }
+
+        $result = AdminUser::where(['account'=>$account])->get();
+
+
+        /**
+         * notes;
+         * cookie里面写入数据，使用"::"分割用户id,名字,学号
+         * 例子:   1::张三::131110192
+         */
+        // step 4.写入session
+        session_start();
+        $_SESSION['name'] = $result[0]->name;
+        $_SESSION['id'] = $result[0]->id;
+        $_SESSION['student_id'] = $result[0]->student_id;
+        setcookie('admin_user',$result[0]->id."::".$result[0]->name."::".$result[0]->student_id,time()+7*24*60*60,'/');
+        return Responder::success('成功');
+    }
+
+
+    /**
+     * @param Request $request
+     * 后台获取名字
+     * @return array
+     */
+    public function getWorkName(){
```

Comment:
```suggestion
    public function getUserName(){
```
这样会好一些，这个 `work` 太中式英语了